util/dataframe_ops.py

```diff
@@ -271,43 +288,219 @@ def parse_check(tolerance_file_name, input_file_ref, input_file_cur, factor):
             - df_cur (pandas.DataFrame): The current DataFrame parsed from the
                                          current input file.
     """
-    df_tol = parse_probtest_csv(tolerance_file_name, index_col=[0, 1])
+    df_tol = parse_probtest_stats(tolerance_file_name, index_col=[0, 1])
 
-    logger.info("applying a factor of %s to the spread", factor)
     df_tol *= factor
 
-    df_ref = parse_probtest_csv(input_file_ref, index_col=[0, 1, 2])
-    df_cur = parse_probtest_csv(input_file_cur, index_col=[0, 1, 2])
-
-    logger.info(
-        "checking %s against %s using tolerances from %s",
-        input_file_cur,
-        input_file_ref,
-        tolerance_file_name,
-    )
+    df_ref = parse_probtest_stats(input_file_ref, index_col=[0, 1, 2])
+    df_cur = parse_probtest_stats(input_file_cur, index_col=[0, 1, 2])
 
     return df_tol, df_ref, df_cur
 
 
-def check_stats_file_with_tolerances(
-    tolerance_file_name, input_file_ref, input_file_cur, factor
+def check_file_with_tolerances(
+    tolerance_file_name, input_file_ref, input_file_cur, factor, rules
```

Comment:
I would use this default value, so you don't need to change the existing use cases.
```suggestion
    tolerance_file_name, input_file_ref, input_file_cur, factor, rules=""
```